lib/vega_lite/data.ex

```diff
@@ -403,42 +403,58 @@ defmodule VegaLite.Data do
         legend_key: legend_key
       )
   """
-  def polar_plot(vl \\ Vl.new(), data, fields, opts \\ []) do
+  def polar_grid(vl \\ Vl.new(), opts \\ []) do
     opts =
       Keyword.validate!(
         opts,
         [
           :radius_marks,
-          :mark,
-          mark_opts: [],
-          return_grid: true,
-          direction: :counter_clockwise,
           angle_marks: [0, 90, 180, 270, 360],
+          direction: :counter_clockwise,
           angle_offset: 0,
-          grid_opacity: 1,
-          grid_color: "white",
-          grid_stroke_color: "black",
-          grid_stroke_opacity: 1,
-          scheme: "turbo"
+          opacity: 1,
+          color: "white",
+          stroke_color: "black",
+          stroke_opacity: 1
         ]
       )
 
-    data_layer = polar_plot_data_layers(data, fields, opts[:mark], opts[:mark_opts], opts)
+    unless opts[:radius_marks] do
+      raise ArgumentError, "missing :radius_marks option"
+    end
+
+    grid_layers = polar_angle_layers(opts)
+    radius_layers = polar_radius_layers(opts)
+
+    vl = append_layers(vl |> Vl.data_from_values(%{"_r" => [0]}), grid_layers ++ radius_layers)
 
-    if opts[:return_grid] do
-      unless opts[:radius_marks] do
-        raise ArgumentError, "missing :radius_marks option"
+    put_in(
+      vl.spec["_vl_polar_config"],
+      opts
+      |> Keyword.take([:radius_marks, :angle_offset, :direction])
+      |> Map.new()
+    )
+  end
+
+  def polar_plot(vl \\ Vl.new(), data, mark, fields, opts \\ []) do
+    opts = Keyword.validate!(opts, scheme: "turbo", hide_axes: false)
```

Comment:
Should we set the scheme? I don't think so.

Reply:
I changed it to be optional

Reply:
It is not about being optional. I don't think we should change the scheme unless the user asks us to. We don't do that for any other the other plots.

Reply:
I've added the color field as an optional meta-field for this configuration. It also is now what affects the legend grouping.